src/org/labkey/test/tests/AdminConsoleTest.java

```diff
@@ -51,6 +59,74 @@ public String getProjectName()
     }
 
     @Test
+    public void testServerHttpHeaderSetting()
+    {
+        goToAdminConsole().clickSiteSettings();
+        waitForElement(Locator.name("includeServerHttpHeader"));
+        WebElement checkbox = Locator.checkboxByName("includeServerHttpHeader").findElement(getDriver());
+
+        boolean originalValue = "true".equals(checkbox.getAttribute("checked"));
+
+        // Try with the setting on
+        if (!originalValue)
+            click(Locator.checkboxByName("includeServerHttpHeader"));
+        clickButton("Save");
+
+        String serverHeader = getServerHeader();
+        assertTrue("Expected to get a Server header, but got " + serverHeader, serverHeader != null && serverHeader.startsWith("LabKey/"));
+
+        // Try with the setting off
+        goToAdminConsole().clickSiteSettings();
+        waitForElement(Locator.name("includeServerHttpHeader"));
+        click(Locator.checkboxByName("includeServerHttpHeader"));
+        clickButton("Save");
+
+        serverHeader = getServerHeader();
+        assertNull("Expected to get no Server header, but got " + serverHeader, serverHeader);
+
+        if (originalValue)
+        {
+            // Turn the setting back on
+            goToAdminConsole().clickSiteSettings();
+            waitForElement(Locator.name("includeServerHttpHeader"));
+            click(Locator.checkboxByName("includeServerHttpHeader"));
+            clickButton("Save");
+        }
+    }
+
+    private static class GetServerHeaderCommand extends SimpleGetCommand
+    {
+        private String _server;
+        public GetServerHeaderCommand()
+        {
+            super("project", "begin");
+        }
+
+        @Override
+        protected Response _execute(Connection connection, String folderPath) throws CommandException, IOException
+        {
+            Response response = super._execute(connection, folderPath);
+            _server = response.getHeaderValue("Server");
+            return response;
+        }
+    }
+
+    @LogMethod(quiet = true)
+    private String getServerHeader()
+    {
+        Connection cn = createDefaultConnection();
+        try
+        {
+            GetServerHeaderCommand command = new GetServerHeaderCommand();
+            command.execute(cn, "/home");
+            return command._server;
+        }
+        catch (IOException | CommandException e)
+        {
+            throw new RuntimeException("Failed to get Server HTTP response header", e);
+        }
+    }
+        @Test
```

Comment:
Spacing looks messed up
```suggestion
    }
    
    @Test
```